libs/util/httpRequest.js

```diff
@@ -135,26 +50,46 @@ function FetchrError(options, request, response, responseBody, originalError) {
     return err;
 }
 
-function io(options) {
+function shouldRetry(options, statusCode, attempt) {
+    if (attempt >= options.retry.maxRetries) {
+        return false;
+    }
+
+    if (options.method === 'POST' && !options.retry.retryOnPost) {
+        return false;
+    }
+
+    return options.retry.statusCodes.indexOf(statusCode) !== -1;
+}
+
+function delayPromise(fn, delay) {
+    return new Promise(function (resolve, reject) {
+        setTimeout(function () {
+            fn().then(resolve, reject);
+        }, delay);
+    });
+}
+
+function io(options, controller) {
     var request = new Request(options.url, {
         body: options.body,
         credentials: options.credentials,
         headers: options.headers,
         method: options.method,
-        signal: options.controller.signal,
+        signal: controller.signal,
     });
 
     var timeoutId = setTimeout(function () {
-        options.controller.abort();
+        controller.abort();
     }, options.timeout);
 
     return fetch(request).then(
         function (response) {
             clearTimeout(timeoutId);
 
             if (response.ok) {
-                return response.text().then(function (responseBody) {
-                    return parseResponse(responseBody);
+                return response.json().catch(function () {
+                    return null;
```

Comment:
return null?

Reply:
Yup :( This is the fetchr behavior since a looong time ago. For me, we could throw an error in this case as well, but I'm not sure yet if we would be fixing or introducing a bug. 

I need to add some functional tests to properly see how fetchr will behave with a 200 response and empty body. I think unless fetchr is not properly set in the server, there is no way to get a empty response...